### PR TITLE
update displayed selected value in select component when an option changes

### LINKED ISSUE #4982: fix: update displayed selected value in select component when an option changes
# 🐛 Bug Report

Given a Select component:

```html
<fast-select>
    <fast-option>This option has no value</fast-option>
    <fast-option disabled>This option is disabled</fast-option>
    <fast-option value="hi">This option has a value</fast-option>
    <fast-option selected>This option is selected by default</fast-option>
</fast-select>
```

If the `innerHTML` of the selected option changes, the displayed value of the select component should reflect the change.

## 💻 Repro or Code Sample

In the current storybook of Select, you can repro by:

```html
<!-- base.html -->
<fast-select>
    <fast-option>This option has no value</fast-option>
    <fast-option disabled>This option is disabled</fast-option>
    <fast-option value="hi">This option has a value</fast-option>
    <fast-option selected class="test">This option is selected by default</fast-option> <!-- add the test class -->
</fast-select>
```

```ts
// select.stories.ts

// add the following lines at the end:
setTimeout(() => {
    (document.querySelector('.test') as HTMLElement).innerHTML = 'Hello world';
}, 1000);
```

End result will look like:

Capture d’écran 2021-07-22 à 21 24 16

where the displayed selected option still displays the "old" innerHTML of the selected option.

## 🤔 Expected Behavior

I'm expecting the change of the option (if it's the one selected) to be reflected in the display of the select. It seems a little "edge case" but this happens in my case with i18n issue which is very important. When a user changes the locale of the website, the localized texts are all automatically changes in realtime, and this should be reflected in the selected option.

## 🔦 Context

As mentioned before, the context is when changing locale in a website or web-application. When a user changes the locale, the localized texts are all automatically changes in realtime, and this should be reflected in the selected option.

## 🌍 Your Environment

Fast Components 2.3.0

### PR DESCRIPTION
**Pull request by radium-v**
# Pull Request

## 📖 Description
Allows the `Select` component to update its display value when the content of its selected options changes.

### 🎫 Issues

Fixes #4982.

## 👩‍💻 Reviewer Notes

The `displayValue` is now a read-only getter, since it was never intended to be manipulated directly.

## 📑 Test Plan

Changing the contents of a selected `<fast-option>` while its parent `<fast-select>` is in single-selection mode should update the displayed value in the dropdown.

### General

- [x] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Review by chrisdholt — APPROVED** [review]
Thanks @radium-v!

File: packages/web-components/fast-foundation/src/select/select.ts [hunk 1]
```diff
@@ -372,6 +378,12 @@ export class Select extends FormAssociatedSelect {
         super.slottedOptionsChanged(prev, next);
         this.setProxyOptions();
         this.updateValue();
+
+        if (this.collapsible) {
+            this.options.forEach(o => {
+                Observable.getNotifier(o).subscribe(this, "content");
```
Comment (EisenbergEffect) [inline packages/web-components/fast-foundation/src/select/select.ts:384]: Whenever there is a manual subscribe call there usually needs to be a corresponding unsubscribe call.

**Review by EisenbergEffect — APPROVED** [review]
I like this approach much better. Nice!

**Review by chrisdholt — APPROVED** [review]
LGTM - Thanks @radium-v!

**Pull request merged**